components/sync_manager/ios/SyncManager/SyncManagerComponent.swift

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import UIKit
+
+open class SyncManagerComponent {
+    private var api: SyncManager
+
+    public init() {
+        api = SyncManager()
+    }
+
+    public func disconnect() {
+        api.disconnect()
+    }
+
+    public func sync(params: SyncParams) throws -> SyncResult {
+        return try api.sync(params: params)
```

Comment:
What do we think about reporting the telemetry here after the sync? that way the caller doesn't have to call reportSyncTelemetry

I could see the argument for the caller having full control over the telemetry though, thoughts?

Reply:
Yeah, that's what I initially attempted. The issue was that when the sync call fails I want to throw and prevent the post-sync code in iOS from being executed. However, when the reporting logic fails, I just want to log the error so the post-sync code isn't bypassed.

This would require either passing an instance of the iOS logger in (which I couldn't do because the logger is defined in that repo) or returning the error message to be logged. That I could do but it would require that the `sync` function throw sync errors and return telemetry reporting errors which seemed to be overloading it and generally in bad form.

I'm open to suggestions here, although it's worth noting when we do eventually move the reporting to the rust layer the `reportSyncTelemetry` function will be obsolete.

Reply:
Oh, I should also add that I didn't want to refactor the iOS code while the sync manager changes are baking in beta.